### ISSUE TITLE
Manually constructed range with start > end

### ISSUE THREAD

**Issue by alice-i-cecile** [rust-lang/rust-clippy#8200]
### What it does

Detects when a user attempted to construct a range using `a..b` notation where `a` is always greater than `b`.

This produces an empty iterator, rather than counting down (as in some other languages).

### Lint Name

range_start_greater_than_range_end

### Category

correctness

### Advantage

The suggested code has either caught a typo where the arguments were swapped, or the user wanted to count down with a range.

### Drawbacks

None. 

### Example

```rust
fn main() {
    for i in 3..0 {
        println!("{}", i);
    }
}
```

will produce an empty iterator, as the start of the range is greater than its end. If you wanted to count down, instead use:

```rust {
fn main() {
    for i in (0..3).rev() {
        println!("{}", i);
    }
}
```

**Comment by llogiq** [comment]
The lint should check if that inverse range is actually iterated. Otherwise, people could trick us into [false positives](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=0e443fe8a6f68950cac0941d0fc17231).

**Comment by giraffate** [comment]
Do you mean the [`reversed_empty_ranges`](https://rust-lang.github.io/rust-clippy/master/index.html#reversed_empty_ranges)?

**Comment by alice-i-cecile** [comment]
Yep, looks like this already exists. I don't know why it didn't catch the mistake 🤔
